### PR TITLE
Update underscore to 1.5.2, change test after _.max to -Infinity.

### PR DESCRIPTION
**Pull request by devkat**

### CODE REVIEW

File: src/compile.js [hunk 1]
```diff
@@ -547,7 +547,7 @@ var compileNodeWithEnvToJsAST = function(n, env, opts) {
                 var maxTagPath = _.max(tagPaths, function(t) {
                     return t.path.length;
                 });
-                var maxPath = maxTagPath ? maxTagPath.path : [];
+                var maxPath = maxTagPath === -Infinity ? [] : maxTagPath.path;
```
Comment (joneshf) [inline src/compile.js:550]: Can you explain briefly why this change is necessary?  I assume that underscore returns `-Infinity` when the array is empty.  Is that correct?

Comment (devkat) [inline src/compile.js:550]: Exacly, this was changed in the meantime:
https://github.com/jashkenas/underscore/commit/348c93515cf56263828c683ebab055e6c800b63b

Comment (joneshf) [inline src/compile.js:550]: Cool.  Thanks.
